metricproviders/influxdb/influxdb.go

```diff
@@ -0,0 +1,129 @@
+package influxdb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/argoproj/argo-rollouts/utils/defaults"
+	"github.com/argoproj/argo-rollouts/utils/evaluate"
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+	influxapi "github.com/influxdata/influxdb-client-go/v2/api"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"time"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	metricutil "github.com/argoproj/argo-rollouts/utils/metric"
+)
+
+const (
+	//ProviderType indicates the provider is InfluxDB
+	ProviderType                    = "Influxdb"
+	DefaultInfluxdbTokensSecretName = "influxdb"
```

Comment:
Maybe a comment here (I tend to comment on all exported types - bit of a hangover from over-zealous vetting tools).

Reply:
Same goes for the one below 👇